lib/chef/resource/cookbook_file.rb

```diff
@@ -27,23 +27,12 @@ class Resource
     class CookbookFile < Chef::Resource::File
       include Chef::Mixin::Securable
 
-      default_action :create
-
-      def initialize(name, run_context = nil)
-        super
-        @provider = Chef::Provider::CookbookFile
-        @source = ::File.basename(name)
-        @cookbook = nil
-      end
+      resource_name :cookbook_file
 
-      def source(source_filename = nil)
-        set_or_return(:source, source_filename, :kind_of => [ String, Array ])
-      end
-
-      def cookbook(cookbook_name = nil)
-        set_or_return(:cookbook, cookbook_name, :kind_of => String)
-      end
+      property :source, [ String, Array ], default: lazy { |r| ::File.basename(r.name) }
```

Comment:
Unless things changed this can be `lazy { ::File.basename(name) }`.

Reply:
Fixed